File: Practica 4/FiguraGeometrica en Python/FiguraGeometrica.py
import math
class FiguraGeometrica:
    def area_circulo(self, radio):
        return math.pi * radio * radio  # Círculo
    def area_rectangulo(self, base, altura):
        return base * altura  # Rectángulo
    def area_triangulo(self, base, altura):
        return (base * altura) / 2.0  # Triángulo
    def area_trapecio(self, base1, base2, altura):
        return ((base1 + base2) * altura) / 2.0  # Trapecio
    def area_pentagono(self, lado, apotema):
        return (5.0 * lado * apotema) / 2.0  # Pentágono
f1 = FiguraGeometrica()
f2 = FiguraGeometrica()
f3 = FiguraGeometrica()
f4 = FiguraGeometrica()
f5 = FiguraGeometrica()
print("Círculo: ", f1.area_circulo(1))
print("Rectángulo: ", f2.area_rectangulo(5.0, 8.0))
print("Triángulo: ", f3.area_triangulo(5, 8))
print("Trapecio: ", f4.area_trapecio(6, 4, 5))
print("Pentágono: ", f5.area_pentagono(5.0, 7))

File: Practica 4/FiguraGeometrica en Java/src/main/java/com/mycompany/figurageometrica/FiguraGeometrica.java
package com.mycompany.figurageometrica;

/**
 * @author Josbel
 */
public class FiguraGeometrica {
    double area(double radio){
        return Math.PI * radio * radio;//circulo
    }
    double area(double base, double altura){
        return base * altura;//rectangulo
    }
    double area(int base, int altura){
        return (base * altura)/2.0;//triangulo
    }
    double area(double base1, double base2, double altura) {
        return ((base1 + base2) * altura) / 2.0; // Trapecio
    } 
    double area(double lado, int apotema) {
        return (5.0 * lado * apotema) / 2.0; // Pentágono
    }

    public static void main(String[] args) {
        FiguraGeometrica f1=new FiguraGeometrica();
        FiguraGeometrica f2=new FiguraGeometrica();
        FiguraGeometrica f3=new FiguraGeometrica();
        FiguraGeometrica f4=new FiguraGeometrica();
        FiguraGeometrica f5=new FiguraGeometrica();
        System.out.println("Circulo: "+f1.area(1));
        System.out.println("Rectangulo: "+f2.area(5.0,8.0));
        System.out.println("Triangulo: "+f3.area(5,8));
        System.out.println("Trapecio: " + f4.area(6, 4, 5));
        System.out.println("Pentágono: " + f5.area(5.0, 7));
    }
}
